calyx-opt/src/passes/well_formed.rs

```diff
@@ -505,6 +563,64 @@ impl Visitor for WellFormed {
         Ok(Action::Continue)
     }
 
+    fn static_invoke(
+        &mut self,
+        s: &mut ir::StaticInvoke,
+        _comp: &mut Component,
+        _ctx: &LibrarySignatures,
+        _comps: &[ir::Component],
+    ) -> VisResult {
+        // Only refers to ports defined in the invoked instance.
+        let cell = s.comp.borrow();
+        let ports: HashSet<_> =
+            cell.ports.iter().map(|p| p.borrow().name).collect();
+
+        s.inputs
+            .iter()
+            .chain(s.outputs.iter())
+            .try_for_each(|(port, _)| {
+                if !ports.contains(port) {
+                    Err(Error::malformed_structure(format!(
+                        "`{}` does not have port named `{}`",
```

Comment:
Can this error ever happen? I thought `from_ast` would take care of ensuring that all ports are defined named on a particular invoke.

Reply:
I don't think it does, although we could change it. To do it would require passing the `SigCtx` for the program as an argument when building the control of the component.

Reply:
ok, changed it.